keps/sig-node/4381-dra-structured-parameters/README.md

```diff
@@ -1411,6 +1449,25 @@ type DeviceRequest struct {
     // +default=false
     AdminAccess bool
 }
+```
+
+Admin access to devices is a privileged operation because it grants users
+access to devices that are in use by other users. Drivers might also remove
+other restrictions when preparing the device.
+
+In Kubernetes 1.31, an example validating admission policy [was
+provided](https://github.com/kubernetes/kubernetes/blob/4aeaf1e99e82da8334c0d6dddd848a194cd44b4f/test/e2e/dra/test-driver/deploy/example/admin-access-policy.yaml#L1-L11)
+which restricts access to this option. It is the responsibility of cluster
+admins to ensure that such a policy is installed if the cluster shouldn't allow
+unrestricted access.
+
+<<[UNRESOLVED pohly ]>>
+
+Do we need to enforce such a policy by default? If so, with which API?
+
+<<[/UNRESOLVED]>>
```

Comment:
This was raised by David Eads when reviewing the VAP that we used in 1.31 (see above).

cc @ritazh

Reply:
Options:
- Keep the *field* as alpha with a separate feature gate (`DRAAdminAccess`) while promoting the rest of the KEP to beta. Defer the decision how to do permission checks.
- Same as before, but describe the field in a separate KEP.
- Implement access control via a `admin-access.kubernetes.io` namespace label (name to be decided) with an in-tree admission controller in 1.32, promote to beta together with this KEP.
- Create a separate `AdminAccessResourceClaim` top-level type and use RBAC permissions to decide who's allowed to create them.

We discussed this in today's WG Device Management meeting and lean towards the first option. A second KEP seemed overkill. But we are aware that this stretches the KEP process a bit. The third option is doable, but it might not be the right API for this kind of access control. Perhaps some future mechanism in Kubernetes will allow doing this in a better way. The last option doesn't scale when the number of claim aspects that need special permission grows.

Reply:
The consensus was to leave the `AdminAccess` field in alpha. This will mean either a separate KEP, or combining that with another set of small enhancements under a single feature gate. Not for any technical reason, but just because our release process doesn't have a way to move KEPs partially to another stage. At least, I think that's true - @jrrickard does that sound right? The question is - could we have two feature gates in one KEP, and have one feature gate advance to beta and the other stay in alpha? I am assuming this will confuse our processes that track features.

Reply:
> combining that with another set of small enhancements under a single feature gate

That feels wrong to me. Each KEP should have its own use cases and proceed according to its own graduation criteria. We don't want to hold back one "sub-feature" because its sibling isn't ready and they share the same feature gate.

Reply:
Added text and feature gate.